### ISSUE TITLE
Remote dropdown dependent on input type not showing

### ISSUE THREAD

**Issue by andrewpros** [selectize/selectize.js#1116]
Hi.

I don't know how the library works internal but i want to make remote search that depends on input type, you can search by name or by product CODE, the code is always and integer, here is my code

```
$('#search-prod').selectize(
        {
            plugins: ['remove_button'],
            loadThrottle: 500,
            valueField: 'id',
            labelField: 'title',
            searchField: 'title',
            options: [],
            create: false,
            persist: false,
            render: {
                option: function (item, escape) {
                    return '<div>' +
                        '<div><span class="title"> ' + escape(item.title) + '</span><span class="jm"> ' + escape(item.jm) + '</span><span class="price"> ' + escape(item.price) + '</span></div>' +
                        '<div class="code">Code ' + escape(item.code) + '</div>' +
                        '</div>';
                }
            },
            load: function (query, callback) {
                if (!query.length) return callback();
                if (query.length < 3) {
                    return;
                }
                $.ajax({
                    url: 'ajaxp.php?q=' + encodeURIComponent(query),
                    type: 'GET',
                    error: function () {
                        callback();
                    },
                    success: function (res) {
                        console.info(res);
                        callback(res);
                    }
                });
            }
        }
    );
```

Now i just checking in php if it is integer or not, if integer i make different query but the resulting json is the same structure.

Now it works when it is text, but when it is integer (the code), the dropdown does not show immediately with result as it does when text entered (json is returned correctly), i need to go out of focus then focus the input again, and then it shows. (also the code has no highlight)

Somebody know why it works that way?

And there is one more problem, the result list are always growing, how to make it reset every time and search again even for the same input text or code?

**Comment by joallard** [comment]
Looks like you erased the placeholder text telling you to read the issue reporting guide (you can also find it in the wiki), make a minimal example, and provide steps to reproduce for us to understand the issue. Please follow the guide and edit your issue accordingly for us to be able to process it.

**Comment by andrewpros** [comment]
What?

This is silly, i don't even know if it is an issue or it is not supported at all.

I have posted the minimum example, this is the whole code, everything is in the description like ELI5.

**Comment by joallard** [comment]
Sorry, but we have a lot of issues to fix and respond, and things to fix and enhance in Selectize, and this is the only format that allows us to be efficient when processing issues. If you find there are ways to improve the issue template and issue guide to be more clear to issue reporters, I'd be really glad to hear about it.

Meanwhile, I need your issue to be as clear as possible: that it takes the least time to (a) figure out the issue you're trying to communicate (why we need steps to reproduce) and (b) to try out your example and go debugging in the least amount of time.

Would you please (a) put your example on Jsfiddle, (b) write down with numbers the exact steps that you do, along with what happens and what you expect to happen, and (c) try to remove as much settings as possible while still keeping the demonstrated behavior (replacing the ajax call with a dummy action would be a good idea)

**Comment by andrewpros** [comment]
It is me again, still not fixed?

Or it is even not supported, i don't know.

Oh, and i can't report the issue with your template cuz it is useless...

It asks "Make sure I'm reporting something precise that needs to be fixed"

AND i don't even now if using one OR another param is supported or not, so how can i know if it needs fixes??? Again, worthless.

If i detect text or id and return values dropdown won't show, the data structure is the same, it is product id or name, but won't show, if it is only id or only name then it shows???

But i can't report it, i can't answer "Make sure I'm reporting something precise that needs to be fixed" cuz i don't know.. silly...

**Comment by joallard** [comment]
Nobody has fixed it, no.

Opening a duplicate issue intentionally is a bad way to get my attention.

Forget about the "Make sure" for a minute, fine.

The important part is: Steps to reproduce. Minimal example.

As said above:

> Would you please (a) put your example on Jsfiddle, (b) write down with numbers the exact steps that you do, along with what happens and what you expect to happen, and (c) try to remove as much settings as possible while still keeping the demonstrated behavior (replacing the ajax call with a dummy action would be a good idea)

Please stay civil when asking for help, after all, you're asking for help from unpaid volunteers.

**Comment by andrewpros** [comment]
Ok, apologize for the reaction.
